wagtail/admin/tests/pages/test_create_page.py

```diff
@@ -73,6 +75,37 @@ def test_add_subpage_with_subpage_types(self):
         # List should not contain page types not in the subpage_types list
         self.assertNotContains(response, "Simple page")
 
+    def test_no_subpage_type_available_to_create(self):
+        # Test if all subpages have reached their max_count_per_parent limit
+        simple_parent_page = SimpleParentPage(
+            title="Hello World!",
+            slug="hello-world",
+        )
+        self.root_page.add_child(instance=simple_parent_page)
+        simple_child_page = SimpleChildPage(
+            title="Hello World!",
+            slug="hello-world",
+        )
+        simple_parent_page.add_child(instance=simple_child_page)
+        BusinessIndex.max_count_per_parent = 1
```

Comment:
Thanks @Jaspreet-singh-1032 - the changes here look good!

I'm just a little concerned about the need to patch `BusinessIndex.max_count_per_parent` here... I can see that it was needed because of [this `subpage_types` rule](https://github.com/wagtail/wagtail/blob/8c2c4a1098522f77b41d231e55619e3713fe67c1/wagtail/test/testapp/models.py#L1609-L1610), which was in turn needed [to stop this existing test from breaking](https://github.com/wagtail/wagtail/blob/main/wagtail/tests/test_tests.py#L273-L278). That was probably the best choice from the perspective of minimising changes to existing tests, but those earlier choices are clearly making maintenance more difficult here.

I'll update the existing tests so that we don't need this workaround, and then merge this.